tajo-core-tests/src/test/java/org/apache/tajo/engine/planner/TestPartitionedTableRewriter.java

```diff
@@ -0,0 +1,152 @@
+import org.apache.tajo.plan.logical.*;
+import org.apache.tajo.plan.rewrite.rules.PartitionedTableRewriter;
+import org.apache.tajo.util.CommonTestingUtil;
+import org.apache.tajo.util.FileUtil;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class TestPartitionedTableRewriter extends QueryTestCaseBase {
+
+  @Test
+  public final void testPartitionPruningUsingManuallyAddedPartition() throws Exception {
+    String tableName = "testPartitionPruningUsingManualAddedPartition".toLowerCase();
+    String canonicalTableName = CatalogUtil.getCanonicalTableName("\"TestPartitionedTableRewriter\"", tableName);
+
+    executeString("create table " + canonicalTableName + " (n_nationkey INT8, n_name TEXT, n_regionkey INT8) " +
+      "partition by column(key TEXT)");
+
+    TableDesc tableDesc = catalog.getTableDesc(getCurrentDatabase(), tableName);
+    assertNotNull(tableDesc);
+
+    FileSystem fs = FileSystem.get(conf);
+    Path path = new Path(tableDesc.getUri().toString() + "/key=part123");
+    fs.mkdirs(path);
+    FileUtil.writeTextToFile("1|ARGENTINA|1", new Path(path, "data"));
+
+    path = new Path(tableDesc.getUri().toString() + "/key=part456");
+    fs.mkdirs(path);
+    FileUtil.writeTextToFile("2|BRAZIL|1", new Path(path, "data"));
+
+    path = new Path(tableDesc.getUri().toString() + "/key=part789");
+    fs.mkdirs(path);
+    FileUtil.writeTextToFile("3|CANADA|1", new Path(path, "data"));
+
+    // Get all partitions
+    verifyFindFilteredPartitionPaths("SELECT * FROM " + canonicalTableName + " ORDER BY key", 3);
+
+    // Get partition with filter condition
+    verifyFindFilteredPartitionPaths("SELECT * FROM " + canonicalTableName + " WHERE key = 'part456' ORDER BY key", 4);
+
+    executeString("DROP TABLE " + canonicalTableName + " PURGE").close();
+  }
+
+  @Test
+  public final void testPartitionPruningUsingDirectories() throws Exception {
+    String tableName = "testPartitionPruningUsingDirectories".toLowerCase();
+    String canonicalTableName = CatalogUtil.getCanonicalTableName("\"TestPartitionedTableRewriter\"", tableName);
+
+    executeString(
+      "create table " + canonicalTableName + "(col1 int4, col2 int4) partition by column(key float8) "
+        + " as select l_orderkey, l_partkey, l_quantity from default.lineitem");
+
+    TableDesc tableDesc = catalog.getTableDesc(getCurrentDatabase(), tableName);
+    assertNotNull(tableDesc);
+
+    // Get all partitions
+    verifyFindFilteredPartitionPaths("SELECT * FROM " + canonicalTableName + " ORDER BY key", 1);
+
+    // Get partition with filter condition
+    verifyFindFilteredPartitionPaths("SELECT * FROM " + canonicalTableName + " WHERE key = 17.0 ORDER BY key", 2);
+
+    executeString("DROP TABLE " + canonicalTableName + " PURGE").close();
+  }
+
+  private void verifyFindFilteredPartitionPaths(String sql, int type) throws Exception {
```

Comment:
```type``` is not defined in anywhere, so it is difficult to understand.
This kind of coding style is very poor and we must avoid. I cannot understand why you try to verify results of different queries in a single method. 
If you want to avoid repeating same code blocks, you should create a method only for those repetitive ones.